x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/esnative/NativeRealmSettings.java

```diff
@@ -17,6 +17,19 @@ public final class NativeRealmSettings {
     public static final String TYPE = "native";
     public static final String DEFAULT_NAME = "default_native";
 
+    /**
+     * This setting is never registered by the security plugin - in order to disable the native user APIs
```

Comment:
I considered it, but had three main reasons for not going down that path:
1. I've removed all User management APIs, which includes reserved users. This means you can't retrieve the `elastic` user, or change its password, etc. That's fine for our needs, but it's likely to break enough things elsewhere in the stack that it could end up being a feature that's too broken for most people's needs
2. If we document it as a real setting then we need to support it, which means deal with the issues above (and more). Is it OK if some unrelated features stop working because they use reserved users? Do we need to invest more time into making this a supported configuration (including testing from other parts of the stack). If we don't document it as a supported setting, then what's the point of making it a real setting?
3. I don't know whether we'll need to keep using these settings forever - particularly in the case of roles. We might decide that we want native role management everywhere, in which case we can drop this internal setting. But it's hard to do that if it was a real setting.

Given the unregistered setting solves our needs, I opted not to open the bigger can of worms. It's worth a discussion as to whether that's a useful feature to put into product, but the current scope of work should be focused on what we really need to achieve.